lib/algos/list/delimiterCurrent.ts

```diff
@@ -9,6 +10,8 @@ type ResultObject = {
     NextMarker ?: string;
 };
 
+const DELIMITER_TIMEOUT_MS = 10 * 1000; // 10s
```

Comment:
I'm curious to understand why 10s, is it related to some kind of limit we already have when making requests?

Reply:
We are working on defining an acceptable approach/value: https://scality.atlassian.net/browse/S3C-7928

Reply:
I see, my 2 cents on this would be to add a scality header to let the client decide, and default to the solution that would ensure expected responses (2nd approach), but that's only my opinion :smile:

Reply:
I am in favor of a limit on the listing, easier to test and levelup module supports the `limit` argument to stop listing after a certain number of entries is returned (https://github.com/Level/levelup#dbcreatereadstreamoptions)